server-side/controllers/user.controller.js

```diff
@@ -88,4 +87,16 @@ export const updatedUser = async (req, res,next) => {
   }
 };
 
-
+export const getUserByEmail = async (req, res, next) => {
+  const email = req.params.email;
+  try {
+    const user = await Users.findOne({ email }).populate({ path: 'visitsWebsites', populate: { path: 'websiteId' } }).populate('profiles preferences').select('-__v');
```

Comment:
access to collections and logics should be in service file.

Reply:
I changed it.